scripts/utils/sign_and_send.js

```diff
@@ -73,7 +67,44 @@ module.exports = function (web3 = web3, artifacts = artifacts) {
     console.log("Transaction awaiting execution in the interface", interfaceLink)
   }
 
+  /**
+   * Checks that a transaction was already proposed to the gnosis-safe UI
+   *
+   * @param {Address} masterSafe Address of the master safe owning the brackets
+   * @param {Transaction} transaction The transaction to be signed and sent
+   * @param {string} network either rinkeby or mainnet
+   * @param {number} [nonce=null] specified transaction index. Will fetch correct value if not specified.
+   */
+  const checkTransactionExistenceOnSafeServer = async function (masterSafe, transaction, network, nonce = null) {
+    const safeTxGas = await estimateGas(masterSafe, transaction)
+    const transactionHash = await masterSafe.getTransactionHash(
+      transaction.to,
+      transaction.value,
+      transaction.data,
+      transaction.operation,
+      safeTxGas,
+      baseGas,
+      0,
+      ZERO_ADDRESS,
+      ZERO_ADDRESS,
+      nonce
```

Comment:
I don't think you can use `nonce = null` to estimate the gas, you need to retrieve the nonce first.

Reply:
Interesting catch... was this detected when running the test plan?

Reply:
Nope, while reading the comments! They say that the nonce must retrieved first, but it's not.